### PR TITLE
Implement LStrInt for ITNative

### PR DESCRIPTION
**Pull request by BartAdv**
LStrInt primitive and newLocal monadic action for unique locals

Added tests in separate case as there's gonna be couple of cases for LStrInt, but I don't yet know if others are applicable at the moment.

### CODE REVIEW

File: src/IRTS/CodegenCil.hs [hunk 1]
```diff
@@ -454,6 +454,20 @@ cgOp LStrTail [v] = do
...
+  val <- newLocal Int32 "val"
+  tell [ ldc_i4 0
+       , stlocN val ]
+  loadString v
+  tell [ ldlocaN val
+       , call [] Bool "mscorlib" "System.Int32" "TryParse" [String, ByRef Int32]
+       , pop
+       , ldlocN val
+       , boxInt32
+       ]
```
Comment (bamboo) [inline src/IRTS/CodegenCil.hs]: Please use the same layout as above with the closing bracket at the same line.

File: src/IRTS/CodegenCil.hs [hunk 2]
```diff
@@ -454,6 +454,20 @@ cgOp LStrTail [v] = do
...
+
+cgOp (LStrInt i) [v] = unsupported "LStrInt" i
```
Comment (bamboo) [inline src/IRTS/CodegenCil.hs]: Please use `_` as the name for unused variable `v`.

File: src/IRTS/CodegenCil.hs [hunk 3]
```diff
@@ -512,6 +526,14 @@ newLabel prefix = do
   put (CodegenState (suffix + 1) locals)
   return $ prefix ++ show suffix
 
+newLocal :: PrimitiveType -> String -> CilCodegen String
+newLocal ty prefix = do
+  (CodegenState suffix locals) <- get
+  let name = prefix ++ show locals
+  tell [ localsInit [ Local ty name]]
```
Comment (bamboo) [inline src/IRTS/CodegenCil.hs]: Spaces missing before both end brackets.

**Comment by BartAdv** [comment]
Should I amend & force push after those changes?

**Comment by bamboo** [comment]
> amend & force push .. ?
:+1:

**Comment by bamboo** [comment]
There seems to  be some mixup with line endings which caused [the build to fail](https://travis-ci.org/bamboo/idris-cil/builds/81664106). Maybe [setting core.autocrlf](https://help.github.com/articles/dealing-with-line-endings/#global-settings-for-line-endings) is in order?

**Comment by BartAdv** [comment]
I see. I will check my Emacs setttings (I'm on windows), don't want to
touch autocrlf (i have it disabled) - it causes much pain:)

**Comment by bamboo** [comment]
Could you rebase it on top of master, please?

**Comment by bamboo** [comment]
nevermind, doing it here.

**Comment by bamboo** [comment]
Merged manually in c095cff65eeeb743299de0921bbb0b51869cb3ea

Excellent job, thanks!

**Pull request closed without merging**